include/reactive_planners/demo_reactive_planners_solo12_step_adjustment.hpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file
+ * @license BSD 3-clause
+ * @copyright Copyright (c) 2023, New York University and Max Planck
+ * Gesellschaft
+ *
+ * @brief A C++ translation of the demos/demo_reactive_planners_solo12_step_adjustment_walk.ipynb script (the first part)
+ *
+ * The 4 fields to control the direction of SOLO are y_des, yaw_des, com_des, and v_des
+ */
+
+#include <Eigen/Dense>
+#include <pinocchio/parsers/urdf.hpp>
+#include <pinocchio/multibody/data.hpp>
+#include "mim_control/impedance_controller.hpp"
+#include "mim_control/centroidal_pd_controller.hpp"
+#include "mim_control/centroidal_force_qp_controller.hpp"
+#include "reactive_planners/quadruped_dcm_reactive_stepper.hpp"
+
+class DemoReactivePlanner {
+public:
+    DemoReactivePlanner();
+
+    /**
+     * Initializes the class
+     * @param path_to_urdf path to SOLO's URDF to build the pinocchio model
+     */
+    DemoReactivePlanner(std::string path_to_urdf);
+
+    /**
+     * Initializes the quadruped_dcm_reactive_stepper
+     * Sets the parameters for some controllers used by the script (e.g., params related to SOLO's direction).
+     * @param q base + joint configuration
+     * @param direction either "forward", "right", "left", "turn_right", "turn_left", "stay" (anything else will be equivalent to stay)
+     */
+    void initialize(Eigen::Matrix<double, 19, 1> &q, std::string direction);
+
+    Eigen::VectorXd
+    /**
+     * Reactive planner script
+     * @param q vector of 19 elements composed on [base position (3 elements), base orientation (4 elements), joint configurations (12 elements)]
+     * @param dq vector of 18 elements composed of [base velocity (3), base ang vel (3), joint velocities (12)]
+     * @param control_time parameter for self.quadruped_dcm_reactive_stepper
+     * @param direction either "forward", "right", "left", "turn_right", "turn_left", "stay" (anything else will be equivalent to stay)
+     * @return torques to send to SOLO
+     */
+    compute_torques(Eigen::Matrix<double, 19, 1> &q, Eigen::Matrix<double, 18, 1> &dq, double control_time, const std::string& direction);
+
+    void quadruped_dcm_reactive_stepper_start();
```

Comment:
We use Doxygen to generate documentation. Please add comments for every function and variable that you defined.

Reply:
+1

Reply:
in vscode there are some tools to help you write the doc for method and attribute